Lib/test/test_io.py

```diff
@@ -4714,7 +4714,8 @@ def _read():
             # handlers, which in this case will invoke alarm_interrupt().
             signal.alarm(1)
             try:
-                self.assertRaises(ZeroDivisionError, wio.write, large_data)
+                with self.assertRaises(ZeroDivisionError):
+                    wio.write(large_data)
```

Comment:
I needed to make this change because otherwise the exception is not swallowed (seems that it shows up after the self.assertRaises call has returned).